src/algebra/pointwise.lean

```diff
@@ -10,23 +10,34 @@ import data.set.finite
 import group_theory.submonoid.basic
 
 /-!
-# Pointwise addition, multiplication, scalar multiplication and vector subtraction of sets.
+# Pointwise operations of sets
 
 This file defines pointwise algebraic operations on sets.
-* For a type `α` with multiplication, multiplication is defined on `set α` by taking
-  `s * t` to be the set of all `x * y` where `x ∈ s` and `y ∈ t`. Similarly for addition.
-* For `α` a semigroup, `set α` is a semigroup.
-* If `α` is a (commutative) monoid, we define an alias `set_semiring α` for `set α`, which then
-  becomes a (commutative) semiring with union as addition and pointwise multiplication as
-  multiplication.
-* For a type `β` with scalar multiplication by another type `α`, this
-  file defines a scalar multiplication of `set β` by `set α` and a separate scalar
-  multiplication of `set β` by `α`.
-* We also define pointwise multiplication on `finset`.
+
+## Main declarations
+
+For sets or finsets `s` and `t` and scalar `a`:
+* `s * t`: Multiplication, set of all `x * y` where `x ∈ s` and `y ∈ t`.
+* `s + t`: Addition, set of all `x + y` where `x ∈ s` and `y ∈ t`.
+* `s⁻¹`: Inversion, set of all `x⁻¹` where `x ∈ s`.
+* `-s`: Negation, set of all `-x` where `x ∈ s`.
+* `s / t`: Division, set of all `x / y` where `x ∈ s` and `y ∈ t`.
+* `s - t`: Subtraction, set of all `x - y` where `x ∈ s` and `y ∈ t`.
+* `s • t`: Scalar multiplication, set of all `x • y` where `x ∈ s` and `y ∈ t`.
+* `s +ᵥ t`: Scalar addition, set of all `x +ᵥ y` where `x ∈ s` and `y ∈ t`.
+* `s -ᵥ t`: Scalar subtraction, set of all `x -ᵥ y` where `x ∈ s` and `y ∈ t`.
+* `a • s`: Scaling, set of all `a • x` where `x ∈ s`.
+* `a +ᵥ s`: Translation, set of all `a +ᵥ x` where `x ∈ s`.
+
+For `α` a semigroup/monoid, `set α` is a semigroup/monoid.
```

Comment:
Do you think this is worth mentioning here?
```suggestion
For `α` a semigroup/monoid, `set α` is a semigroup/monoid.
As an unfortunate side effect, this means that `n • s` where `n : ℕ` is ambiguous between pointwise
scaling and repeated pointwise addition; the former has `(2 : ℕ) • {1, 2} = {2, 4}`, while the
latter has `(2 : ℕ) • {1, 2} = {2, 3, 4}`.
```

Reply:
Yes definitely!